src/Nethermind/Nethermind.Evm/CodeAnalysis/JumpdestAnalyzer.cs

```diff
@@ -13,73 +13,64 @@
 // 
 //  You should have received a copy of the GNU Lesser General Public License
 //  along with the Nethermind. If not, see <http://www.gnu.org/licenses/>.
+// 
 
-using System;
 using System.Collections;
+using System.Threading;
 using Nethermind.Evm.Precompiles;
 
-namespace Nethermind.Evm
+namespace Nethermind.Evm.CodeAnalysis
 {
-    public class CodeInfo
+    public class JumpdestAnalyzer : ICodeInfoAnalyzer
     {
-        private BitArray _validJumpDestinations;
-        private BitArray _validJumpSubDestinations;
+        private byte[] MachineCode { get; set; }
 
-        public CodeInfo(byte[] code)
-        {
-            MachineCode = code;
-        }
+        private BitArray? _validJumpDestinations;
+        private BitArray? _validJumpSubDestinations;
 
-        public bool IsPrecompile => Precompile != null;
-        
-        public CodeInfo(IPrecompile precompile)
+        public JumpdestAnalyzer(byte[] code)
         {
-            Precompile = precompile;
-            MachineCode = Array.Empty<byte>();
+            MachineCode = code;
         }
-        
-        public byte[] MachineCode { get; set; }
-        public IPrecompile Precompile { get; set; }
-
+    
         public bool ValidateJump(int destination, bool isSubroutine)
         {
             if (_validJumpDestinations is null)
             {
                 CalculateJumpDestinations();
             }
-
-            if (destination < 0 || destination >= MachineCode.Length ||
+            
+            if (destination < 0 || destination >= _validJumpDestinations.Length ||
                 (isSubroutine ? !_validJumpSubDestinations.Get(destination) : !_validJumpDestinations.Get(destination)))
             {
                 return false;
             }
 
             return true;
         }
-      
+
         private void CalculateJumpDestinations()
         {
             _validJumpDestinations = new BitArray(MachineCode.Length);
             _validJumpSubDestinations = new BitArray(MachineCode.Length);
-            
+
             int index = 0;
             while (index < MachineCode.Length)
             {
-                //Instruction instruction = (Instruction)code[index];
-                byte instruction = MachineCode[index];                
-                
-                //if (instruction == Instruction.JUMPDEST
+                byte instruction = MachineCode[index];
+
+                // JUMPDEST
                 if (instruction == 0x5b)
```

Comment:
btw why arn't we using Instruction enum here? Would be more readable.

Reply:
no, it is a microoptimization (it was my code)

Reply:
5% faster, the commented code stays